jquery.initialize.js

```diff
@@ -34,26 +38,51 @@
         $(selector).each(callbackOnce);
 
         // Then, add it to the list of selector observers.
-        this.push(new MutationSelectorObserver(selector, callbackOnce));
+        this.push(new MutationSelectorObserver(selector, callbackOnce, options));
     };
 
     // The MutationObserver watches for when new elements are added to the DOM.
     var observer = new MutationObserver(function (mutations) {
 
         // For each MutationSelectorObserver currently registered.
         for (var j = 0; j < msobservers.length; j++) {
-            $(msobservers[j].selector).each(msobservers[j].callback);
+            if (msobservers[j].options.scanDocument) {
+
+                // Scan the entire document.
+                $(msobservers[j].selector)
+                    .each(msobservers[j].callback);
+            } else {
+
+                // For each mutation.
+                for (var m = 0; m < mutations.length; m++) {
+                    // If mutation type is observed.
+                    if ($.inArray(mutations[m].type, mtypes) != -1) {
```

Comment:
For code I'd consider negative `ifs` eg
```
if (!$.inArray(mutations[m].type, mtypes) != -1) {
  continue;
}
// rest of the code. HEY! we caved some tabs in indent!
```

Reply:
I will update accordingly as I implement your other concerns.

Reply:
Don't forget to add yourself to contributors as well as some note in the README